apps/dotcms-ui/src/app/portlets/shared/dot-content-types-listing/components/dot-content-type-clone-dialog/dot-content-type-clone-dialog.component.spec.ts

```diff
@@ -0,0 +1,133 @@
+import { SiteService } from '@dotcms/dotcms-js';
+import { SiteServiceMock } from '@tests/site-service.mock';
+import { DotFormSelectorModule } from '@portlets/dot-edit-page/content/components/dot-form-selector/dot-form-selector.module';
+import { DOTTestBed } from '@dotcms/app/test/dot-test-bed';
+import { DotDialogModule } from '@components/dot-dialog/dot-dialog.module';
+
+@Component({
+    selector: 'dot-test-host-component',
+    template: `
+        <dot-content-type-clone-dialog
+            [isVisibleDialog]="isVisibleDialog"
+            [isSaving]="isSaving"
+        ></dot-content-type-clone-dialog>
+    `
+})
+class TestHostComponent {
+    @Input() isVisibleDialog: boolean;
+    @Input() isSaving: boolean;
+}
+
+const formValues = {
+    name: 'Name of the copied content type',
+    variable: 'variable-name',
+    folder: '',
+    host: '',
+    icon: ''
+};
+
+describe('DotContentTypeCloneDialogComponent', () => {
+    const siteServiceMock = new SiteServiceMock();
+    let component: DotContentTypeCloneDialogComponent;
+    let fixture: ComponentFixture<TestHostComponent>;
+    let de: DebugElement;
+    let dotdialog: DebugElement;
+
+    beforeEach(
+        waitForAsync(() => {
+            DOTTestBed.configureTestingModule({
+                declarations: [DotContentTypeCloneDialogComponent, TestHostComponent],
+                imports: [
+                    DotFormSelectorModule,
+                    BrowserAnimationsModule,
+                    DotFieldValidationMessageModule,
+                    DotMdIconSelectorModule,
+                    SiteSelectorFieldModule,
+                    DotDialogModule
+                ],
+                providers: [{ provide: SiteService, useValue: siteServiceMock }]
+                //schemas: [CUSTOM_ELEMENTS_SCHEMA]
+            });
+            fixture = DOTTestBed.createComponent(TestHostComponent);
+            de = fixture.debugElement.query(By.css('dot-content-type-clone-dialog'));
+            component = de.componentInstance;
+            fixture.detectChanges();
+            dotdialog = de.query(By.css('dot-dialog'));
+
+            component.isVisibleDialog = true;
+            fixture.detectChanges();
+        })
+    );
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+    it('should have a form', () => {
+        fixture.detectChanges();
+        const form: DebugElement = de.query(By.css('form'));
+        expect(form).not.toBeNull();
+        expect(component.form).toEqual(form.componentInstance.form);
+    });
+    it('should be invalid if no name was added', () => {
+        fixture.detectChanges();
+        expect(component.form.valid).toEqual(false);
+    });
+    it('should be valid if bundle field is added', () => {
+        fixture.detectChanges();
+
+        component.form.setValue(formValues);
+        expect(component.form.valid).toEqual(true);
+    });
+    it('should call cancelBtn() on cancel button click', async () => {
```

Comment:
instead of test the call of methods of the component, for example n this case, test the emit of the input in the `dot-test-host-component`.